### PR TITLE
refactor(DB/Misc): Improve loading time

### PR DESCRIPTION
**Pull request by Kitzunu**
* NOTE: Running these SQL files can take several minutes (up to 30minutes depending on your system!)

* This will improve loading time over 50%

## Changes Proposed:
* NOTE: Running these SQL files can take several minutes (up to 30minutes depending on your system!)
* It can feel like the updater gets stuck or freezses. It doesnt. Just a lot of queries. A small price to pay one time for a big improvement in performance

On my system it took about 3-4 minutes to run it

-  Update engine, collation and get rid of MEDIUMINT in all databases
-  These changes will cut loading worldserver by over 50%

Before these changes
`WORLD: World Initialized In 0 Minutes 18 Seconds`

After these changes
`WORLD: World Initialized In 0 Minutes 7 Seconds`

## SOURCE:

Inspired by https://github.com/TrinityCore/TrinityCore/commit/c102baf99e1640df72a25444e3f206e1920a0b18
## Tests Performed:

- Tested on local machiene. Huge improvement in loading times.

## How to Test the Changes:

1. Import the changes with the autoupdater (Risk is that using i.e Heidi it will crash du to Heidi being mediocre)
2. See performance bonus 😎 

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Review by acidmanifesto — APPROVED** [review]

**Comment by Winfidonarleyan** [comment]
- Before PR 7-8 secs
-

**Comment by Kitzunu** [comment]
So now the question is, do we value MySQL or MariaDB more

**Comment by Gultask** [comment]
@noisiver Would you be interested in trying as well?

**Comment by noisiver** [comment]
Keep in mind that my MariaDB server (version 10.6.11) dedicated to AzerothCore runs with a very low amount of resources so performance is ass when compared to something decent but you can see the difference in times.

Before: 15-18 seconds
After: 22-23 seconds

I don't really care about a few seconds. Assuming it does in fact help with startup times for MySQL and those seconds were so important to me I would just switch to MySQL. I'm not going to try using MySQL though.

**Comment by Kitzunu** [comment]
I will put To Be Merged as I see the gain as greater than the loss to MariaDB

Also the vast majority of AC maintainers use MySQL

**Comment by noisiver** [comment]
> I will put To Be Merged as I see the gain as greater than the loss to MariaDB
> 
> Also the vast majority of AC maintainers use MySQL

I agree. If someone feels it's so important to save those seconds they can switch to MySQL. We're not talking minutes here after all so the loss is (in my opinion) minimal.

File: data/sql/updates/pending_db_world/rev_1679045387127819800.sql [hunk 1]
```diff
@@ -0,0 +1,669 @@
...
+ALTER TABLE `creature_equip_template` MODIFY `ItemID3` INT UNSIGNED NOT NULL DEFAULT '0';
+ALTER TABLE `creature_equip_template` MODIFY `VerifiedBuild` INT DEFAULT NULL;
+ALTER TABLE `creature_loot_template` MODIFY `Entry` INT UNSIGNED NOT NULL DEFAULT '0';
+ALTER TABLE `creature_loot_template` MODIFY `Item` INT UNSIGNED NOT NULL DEFAULT '0';
+ALTER TABLE `creature_loot_template` MODIFY `Reference` INT NOT NULL DEFAULT '0';
+ALTER TABLE `creature_model_info` MODIFY `DisplayID` INT UNSIGNED NOT NULL DEFAULT '0';
+ALTER TABLE `creature_model_info` MODIFY `DisplayID_Other_Gender` INT UNSIGNED NOT NULL DEFAULT '0';
+ALTER TABLE `creature_onkill_reputation` MODIFY `creature_id` INT UNSIGNED NOT NULL DEFAULT '0' COMMENT 'Creature Identifier';
+ALTER TABLE `creature_onkill_reputation` MODIFY `RewOnKillRepValue1` INT NOT NULL DEFAULT '0';
+ALTER TABLE `creature_onkill_reputation` MODIFY `RewOnKillRepValue2` INT NOT NULL DEFAULT '0';
```
Comment (Gultask) [inline data/sql/updates/pending_db_world/rev_1679045387127819800.sql:63]: https://github.com/azerothcore/azerothcore-wotlk/pull/15818/files

**Review by Gultask — CHANGES_REQUESTED** [review]

File: data/sql/updates/pending_db_world/rev_1679045387127819800.sql (hunk 1 above)
Comment (Gultask) [inline data/sql/updates/pending_db_world/rev_1679045387127819800.sql:63]:
```suggestion
```

**Review by Gultask — CHANGES_REQUESTED** [review]
https://github.com/azerothcore/azerothcore-wotlk/pull/14099

**Comment by Kitzunu** [comment]
Yea I dont need to update anything for it.

**Review by Gultask — APPROVED** [review]

**Pull request merged**